### PR TITLE
Fixes timestop! (and tweaks its sounds)

### LINKED ISSUE #28548: Wizard timestop doesn't freeze mobs
The overlay stayed for a very long time as well

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
fix: Time stop is now fixed, finally!
soundadd: Time stop's sound now plays in reverse when the effect ends.
/:cl:

Fixes #28548

Additionally, timestop's sound now plays in reverse when it ends, and the sounds are all a bit quieter.

### CODE REVIEW

**Comment by Repukan** [comment]
>Bit quieter

Shit change, make them louder

**Comment by Xhuis** [comment]
[You mean like this?](https://www.dropbox.com/s/eugjutozsmh0a2m/new_sound.ogg?dl=0)

File: code/modules/research/xenobiology/xenobiology.dm [hunk 1]
```diff
@@ -563,16 +563,19 @@
...
 /obj/effect/timestop/Initialize()
 	. = ..()
-	for(var/M in GLOB.living_mob_list)
+	for(var/M in GLOB.player_list)
 		var/mob/living/L = M
-		for(var/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop/T in L.mind.spell_list) //People who can stop time are immune to timestop
-			immune |= L
+		for(var/V in L.mind.spell_list) //People who can stop time are immune to its effects
+			var/obj/O = V
+			if(O.type == /obj/effect/proc_holder/spell/aoe_turf/conjure/timestop)
```
Comment (RemieRichards) [inline code/modules/research/xenobiology/xenobiology.dm]: Come on.

**Review by RemieRichards — DISMISSED** [review]

File: code/modules/research/xenobiology/xenobiology.dm (hunk 1 above)
Comment (Xhuis) [inline code/modules/research/xenobiology/xenobiology.dm]: Oh my god I'm an idiot

File: code/modules/research/xenobiology/xenobiology.dm [hunk 2]
```diff
@@ -563,16 +563,19 @@
...
+			if(istype(O, /obj/effect/proc_holder/spell/aoe_turf/conjure/timestop))
+				immune += L
+				continue
 	timestop()
 
 
 /obj/effect/timestop/proc/timestop()
 	set waitfor = FALSE
-	playsound(get_turf(src), 'sound/magic/timeparadox2.ogg', 100, 1, -1)
+	playsound(get_turf(src), 'sound/magic/timeparadox2.ogg', 75, 1, -1)
```
Comment (Cyberboss) [inline code/modules/research/xenobiology/xenobiology.dm]: playsound(src, ...

File: code/modules/research/xenobiology/xenobiology.dm [hunk 3]
```diff
@@ -598,6 +601,7 @@
 		stoplag()
 
 	//End
+	playsound(get_turf(src), 'sound/magic/timeparadox2.ogg', 75, TRUE, frequency = -1) //reverse!
```
Comment (Cyberboss) [inline code/modules/research/xenobiology/xenobiology.dm]: playsound(src, ...

File: code/modules/research/xenobiology/xenobiology.dm [hunk 4]
```diff
@@ -563,16 +563,19 @@
...
 /obj/effect/timestop/Initialize()
 	. = ..()
-	for(var/M in GLOB.living_mob_list)
+	for(var/M in GLOB.player_list)
 		var/mob/living/L = M
-		for(var/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop/T in L.mind.spell_list) //People who can stop time are immune to timestop
-			immune |= L
+		for(var/V in L.mind.spell_list) //People who can stop time are immune to its effects
+			var/obj/O = V
+			if(istype(O, /obj/effect/proc_holder/spell/aoe_turf/conjure/timestop))
```
Comment (Cyberboss) [inline code/modules/research/xenobiology/xenobiology.dm]: just do `if(locate(/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop) in L.mind.spell_list)`

File: code/modules/research/xenobiology/xenobiology.dm [hunk 5]
```diff
@@ -563,16 +563,19 @@
 
 /obj/effect/timestop/Initialize()
 	. = ..()
-	for(var/M in GLOB.living_mob_list)
+	for(var/M in GLOB.player_list)
 		var/mob/living/L = M
-		for(var/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop/T in L.mind.spell_list) //People who can stop time are immune to timestop
-			immune |= L
+		for(var/V in L.mind.spell_list) //People who can stop time are immune to its effects
```
Comment (Cyberboss) [inline code/modules/research/xenobiology/xenobiology.dm]: Need to check that mind exists

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/research/xenobiology/xenobiology.dm (hunk 5 above)
Comment (Xhuis) [inline code/modules/research/xenobiology/xenobiology.dm]: Because this is checking `GLOB.player_list` a mind should always exist, shouldn't it?

Comment (Cyberboss) [inline code/modules/research/xenobiology/xenobiology.dm]: Hmm you're probably right

File: code/modules/research/xenobiology/xenobiology.dm [hunk 6]
```diff
@@ -563,16 +563,17 @@
...
 /obj/effect/timestop/Initialize()
 	. = ..()
-	for(var/M in GLOB.living_mob_list)
+	for(var/M in GLOB.player_list)
 		var/mob/living/L = M
-		for(var/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop/T in L.mind.spell_list) //People who can stop time are immune to timestop
-			immune |= L
+		if(locate(/obj/effect/proc_holder/spell/aoe_turf/conjure/timestop) in L.mind.spell_list) //People who can stop time are immune to its effects
+			immune += L
+			continue
```
Comment (Cyberboss) [inline code/modules/research/xenobiology/xenobiology.dm]: This continue is unecessary

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**